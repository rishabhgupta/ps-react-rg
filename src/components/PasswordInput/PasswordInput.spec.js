import React from 'react';
import renderer from 'react-test-renderer';
import PasswordInput from './PasswordInput';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

test('hides password quality bar by default', () => {
   const tree = renderer.create(<PasswordInput htmlId="test" name="test" onChange={() => {}} value="asdasdasdasd"/>).toJSON();
   expect(tree).toMatchSnapshot();
});

test('toggles input type when show/hide password clicked', () => {
   const wrapper = shallow(
       <PasswordInput
           htmlId="test"
           name="test"
           onChange={() => {}}
           showVisibilityToggle
           value=""
       />
   );
   expect(wrapper.find({type: 'password'})).toHaveLength(1);
   expect(wrapper.find({type: 'text'})).toHaveLength(0);

   wrapper.find('a').simulate('click');

   expect(wrapper.find({type: 'password'})).toHaveLength(0);
   expect(wrapper.find({type: 'text'})).toHaveLength(1);
});
