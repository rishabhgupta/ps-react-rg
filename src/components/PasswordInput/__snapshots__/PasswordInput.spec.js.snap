// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hides password quality bar by default 1`] = `
<div
  style={
    Object {
      "marginBottom": 16,
    }
  }
>
  <label
    htmlFor="test"
    style={
      Object {
        "display": "block",
      }
    }
  >
    Password
     
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
       *
    </span>
  </label>
  <input
    id="test"
    maxLength={50}
    name="test"
    onChange={[Function]}
    type="password"
    value="asdasdasdasd"
  />
</div>
`;
